lms/static/scripts/frontend_apps/utils/api.ts

```diff
@@ -350,15 +362,24 @@ export function usePaginatedAPIFetch<
     }
   }, [fetchResult.data?.pagination.next, fetchResult.isLoading]);
 
-  // Every time data is loaded, append the result to the list
   useEffect(() => {
+    // Every time data is loaded, append the result to the list
     setCurrentList(prev => {
       if (!fetchResult.data) {
         return prev;
       }
 
       return [...(prev ?? []), ...fetchResult.data[prop]] as ListType;
     });
+
+    const hasMorePages = !!fetchResult.data?.pagination.next;
+    setHasMorePages(hasMorePages);
+
+    // Set pageSize if the API indicated there are more pages.
+    // Once page size has been set, we keep the value, ignoring next page loads
```

Comment:
This comment is out of date, as the page size is updated after each non-final page is fetched, although it will usually always be the same value. I would suggest that you simply remove the comment.

Reply:
This comment is meant to explain the fact that we do `prev => prev ?? fetchResult.data?.[prop].length`, so once `prev` is set we keep it, but perhaps it's self explanatory, or otherwise, it should be move right above that line.

Reply:
Right. Ignore my comment here, you can leave this as-is.